### DISCUSSION TITLE
JSON to JSON transformation

### DISCUSSION THREAD

**Discussion by VelSankar-DT** [apache/camel-karavan#1050] [Q&A] [answered]
How to do Json to Json transformation in apache camel karavan?
I want to generate a new message with an existing message and some extra fields. Like wso2 payload factory mediator. How we achieve it in our apache camel karavan ?

**Comment by mgubaidullin** [comment, marked as answer]
I use this one:
1. Use `Unmarshall` to convert JSON to Object
2. Create output Object using `setBody` and Groovy Expression
3. User `Marshall` to convert Object to JSON

Another option: https://github.com/apache/camel-kamelets-examples/tree/main/jbang/json-transform

**Reply by VelSankar-DT** [reply]
I tried out the GitHub example you suggested, and it's working fine! Thanks a bunch for sharing.
